.github/workflows/reusable-tag-cleanup-commit.yaml

```diff
@@ -0,0 +1,99 @@
+name: Tag Cleanup for Commits
+on:
+  workflow_call:
+    inputs:
+      number-to-keep:
+        required: false
+        type: string
+        default: 3
+    secrets:
+      namespace:
+        required: true
+      openshift-api:
+        required: true
+      openshift-token:
+        required: true
+
+jobs:
+  image-stream-tag-cleanup:
+    name: Image Stream Tag Cleanup
+    runs-on: ubuntu-latest
+    steps:
+    - name: Log in to OpenShift
+      uses: redhat-actions/oc-login@v1
+      with:
+        namespace: ${{ secrets.namespace }}
+        openshift_server_url: ${{ secrets.openshift-api }}
+        openshift_token: ${{ secrets.openshift-token }}
+
+    - name: Image Stream Tag Cleanup
+      run: |
+        NAMESPACE=${{ secrets.namespace }}
+        NUMBER_TO_KEEP=${{ inputs.number-to-keep }}
+
+        IMAGE_STREAM_TAGS=$(oc -n $NAMESPACE get imagestreamtags -o json | jq -r \
+          '.items[] | "\(.image.metadata.name),\(.metadata.creationTimestamp),\(.metadata.name)"' | \
+          egrep ':commit-[0-9a-f]{7}$')
+
+        cleanup_imagestream () {
+          ITEMS=$*
+
+          if [ -z "$ITEMS" ]; then
+            return
+          fi
+
+          COUNT=$(echo $ITEMS | wc -w)
+          if [ $COUNT -le $NUMBER_TO_KEEP ]; then
+            for ITEM in $ITEMS; do
+              TAG=$(echo $ITEM | cut -d, -f2,3 | sed 's/,/:/')
+              echo Not deleting imagestreamtag $TAG because we keep the last \
+                $NUMBER_TO_KEEP in case we want to roll back.
+            done
+
+            return
+          fi
+
+          ITEMS=$(echo $ITEMS | sed 's/ /\n/g' | sort -n | \
+            head -n -$NUMBER_TO_KEEP)
+
+          for ITEM in $ITEMS; do
+            TAG=$(echo $ITEM | cut -d, -f2,3 | sed 's/,/:/')
+            COMMAND="oc -n $NAMESPACE delete imagestreamtag $TAG"
+            echo $COMMAND
+            # $COMMAND
```

Comment:
There is no way for us to truly test GitHub Actions before merging them, so the code in this PR will only _print_ the command to be run - no imagestream tags will be deleted until we are sure it's doing what we want.